### PR TITLE
Create VarLife.md

### PR DESCRIPTION
**Pull request by elendiastarman**
Topics covered: OTCA metapixel, VarLife, basic VarLife circuitry.

### CODE REVIEW

File: VarLife.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+## OTCA Metapixel
+
+<img src="http://www.conwaylife.com/w/images/3/3a/Otcametapixel.png"/>
+(Source: [http://www.conwaylife.com/w/images/3/3a/Otcametapixel.png](http://www.conwaylife.com/w/images/3/3a/Otcametapixel.png))
```
Comment (Mego) [inline VarLife.md]: `[Source](http://www.conwaylife.com/w/images/3/3a/Otcametapixel.png)` would be better, or just leave it off because the next link references the source page on the wiki.

File: VarLife.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+> The OTCA metapixel is a 2048 × 2048 period 35328 unit cell that was constructed by Brice Due... It has many advantages...including the ability to emulate any Life-like cellular automaton and the fact that, when zoomed out, the ON and OFF cells are easy to distinguish...
+
+What [Life-like cellular automata](http://www.conwaylife.com/wiki/Life-like_cellular_automaton#Life-like_cellular_automata) means here is essentially that cells are born and cells survive according to how many of their eight neighbor cells are alive. The syntax for these rules is as follows: a B followed by the numbers of live neighbors that will cause a birth, then a slash, then an S followed by the numbers of live neighbors that will keep the cell alive. A bit wordy, so I think an example will help. The canonical Game of Life can be represented by the rule B23/S3, which says that any dead cell with two or three live neighbors will become alive and any live cell with three neighbors will remain alive. Otherwise, the cell dies.
+
```
Comment (Mego) [inline VarLife.md:11]: You might want to go a bit more in-depth on the OTCA metapixel here - the overlap for adjacent metapixels, how the rules are set, how neighbor detection works, etc.

File: VarLife.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+[insert gif]
+
+Diagonal wires also exist but they are not used much at all.
+
+[insert gif]
+
+### Gates
+
+### Delay Components
```
Comment (Mego) [inline VarLife.md]: The headers aren't present yet - are you planning on talking about stuff like memory cells? That might overlap a bit with the part that @ElectroRedstoner is writing, which may or may not be desirable.

**Comment by ElectroRedstoner** [comment]
In terms of hardware, I talked about the stuff I did: The memory cells, RAM, demultiplexer, ROM, etc. I think we should put the things you guys started with, like the 11x11 tiles, logic gates, and delay, into this write up.

File: VarLife.md (hunk 1 above)
Comment (elendiastarman) [reply VarLife.md]: When I make the actual post, I'm planning on hosting all images with SE's Imgur, so I want to keep a link to the original. That said, I like the idea of linking just the word "Source".

File: VarLife.md (hunk 2 above)
Comment (elendiastarman) [reply VarLife.md:11]: The LifeWiki page does that though. I didn't feel those details were very relevant to our work; it's a sort of implementation detail. If you think otherwise, I suppose I can summarize what the LifeWiki page says.

Comment (Mego) [reply VarLife.md:11]: Those details are important in the generation of GoL patterns from Varlife patterns, so they are relevant to our work. I'm not suggesting an essay on them - a summary will be sufficient.

File: VarLife.md [hunk 4]
```diff
@@ -60,4 +60,22 @@ Short url: http://play.starmaninnovations.com/varlife/kJotsdSXIj
 
 ### Gates
 
+There are actually a lot of ways to construct each individual gate, so I will only be showing one example of each kind. This first gif demonstrates AND, XOR, and OR gates, respectively. The basic idea here is that a green cell acts like an AND, a blue cell acts like an XOR, and a red cell acts like an OR, and all the other cells around them are just there to control the flow properly.
+
+[Remember to upload to SE Imgur]  
+<img src="http://play.starmaninnovations.com/static/d3applets/renders/ixoJIHLDPe.gif"/>  
+Short url: http://play.starmaninnovations.com/varlife/EGTlKktmeI
+
+[[Question for reviewers: should I include the ANT tile as seen in KZhang's hardware post?]]
```
Comment (Mego) [inline VarLife.md]: I think it's fine either way.

File: VarLife.md [hunk 5]
```diff
@@ -9,6 +9,12 @@ The [OTCA Metapixel](http://www.conwaylife.com/w/index.php?title=OTCA_metapixel)
...
 
+Despite being a 2048 x 2048 cell, the OTCA metapixel actually has a bounding box of 2058 x 2058 cells, the reason being that it overlaps by five cells in every direction with its neighbors. [[Why? What does this do? I have no idea.]] The birth and survival rules are encoded in a special section of cells at the left side of the metapixel, by the presence or absence of eaters in specific positions along two columns (one for birth, the other for survival). As for detecting the state of neighboring cells, here's how that happens:
```
Comment (Mego) [inline VarLife.md]: The overlap occurs because it is necessary for correct timing and neighbor metapixel updates. Without the overlap, the timing would be more complex, and would likely cause the metapixel to be larger.

Comment (ElectroRedstoner) [reply VarLife.md]: More specifically, the overlap is only on the diagonal neighbors. When a metapixel turns on, it will shoot out a glider to signal its neighbors that it's on. However, if there is no neighbor, that glider must be eaten, or else that glider can fly off indefinitely or break other metapixels. The overlap is necessary for the detection of the diagonal neighbors, but not for the orthogonal neighbors.

**Review by ElectroRedstoner — DISMISSED** [review]
The ANT gate is a gate that passes a signal from A if and only if there is no signal from B.

**Review by Mego — APPROVED** [review]
Looks good to me!
